pyht/telemetry.py

```diff
@@ -0,0 +1,170 @@
+from __future__ import annotations
+
+import time
+
+
+class Telemetry:
+    def __init__(self, buffer_size: int = 1000):
+        self._metrics = []
+        self._buffer_size = buffer_size
+
+    def start(self, operation: str) -> Metrics:
+        metrics = Metrics()
+        metrics.start(operation)
+        if len(self._metrics) >= self._buffer_size:
+            self._metrics.pop(0)
+        self._metrics.append(metrics)
+        return metrics
+
+    def metrics(self) -> list[Metrics]:
+        return self._metrics
+
+
+class Metrics:
+    """
+      {
+        "operation": "tts-request",
+        "status": "ok",
+        "startTime": 1712937515393,
+        "endTime": 1712937515442,
+        "duration": 0.679,
+        "counters": {
+          "chunk": 10,
+          "ok": 1,
+          "error": 0
+        },
+        "attributes": {
+          "endpoint": [
+            "vocode-000000000000002a.on-prem.play.ht:11045"
+          ],
+          "text": "This is a test.  This is the second sentence."
+        },
+        "timers": {
+          "time-to-first-audio": {
+            "duration": 0.149,
+            "count": 1
+          },
+          "tts-request": {
+            "duration": 0.679,
+            "count": 1
+          }
+        }
+      }
+    """
+
+    def __init__(self):
+        self.operation = None
+        self.status = None
+        self.start_time = None
+        self.end_time = None
+        self.duration = None
+        self.counters = {}
+        self.attributes = {}
+        self.timers = {}
+
+    def start(self, operation: str) -> Metrics:
+        self.operation = operation
+        self.start_time = time.time()
+        print(f"Start {operation} @ {self.start_time}")
+        return self
+
+    def inc(self, counter: str, count: int = 1) -> Metrics:
+        if counter not in self.counters:
+            self.counters[counter] = 0
+        self.counters[counter] += count
+        return self
```

Comment:
nit: prefer `.get()` or `.setdefault()` for dict access.
```python
def inc(self, counter: str, count: int = 1) -> 'Metrics':  # quote return type of self for backwards compat.
    self.counters[counter] = self.counters.get(counter, 0) + count
    return self
```

`.setdefault()` is for ref objects
```python
class Counter:
    def __init__(self, value: int = 0):
        self._v = value
    def __repr__(self):
        return repr(self._v)
    def __add__(self, v: int):
        self._v += v
        return self
    def __sub__(self, v: int):
        self._v -= v
        return self

def inc(self, counter: str, count: int = 1) -> 'Metrics':
    self.counters.setdefault(counter, Counter()) + 1
    return self
```

Reply:
Fixed.